src/tribler/core/utilities/pony_utils.py

```diff
@@ -281,27 +378,78 @@ def acquire_lock(self):
             info.lock_wait_total_duration += lock_wait_duration
 
     def release_lock(self):
+        # Adds tracking of a db_session's total lock hold duration
         super().release_lock()
         info = local.db_session_info
         if info is not None:
             acquire_time = self._acquire_time
             lock_hold_duration = time.time() - acquire_time
             info.lock_hold_total_duration += lock_hold_duration
 
+    def set_transaction_mode(self, connection, cache):
+        with marking_corrupted_db(self.pool.filename):
+            return super().set_transaction_mode(connection, cache)
+
+    def execute(self, cursor, sql, arguments=None, returning_id=False):
+        with marking_corrupted_db(self.pool.filename):
+            return super().execute(cursor, sql, arguments, returning_id)
+
+    def mark_db_as_malformed(self):
+        filename = self.pool.filename
+        if not Path(filename).exists():
+            raise RuntimeError(f'Corrupted database file not found: {filename!r}')
+
+        marker_filename = filename + '.is_corrupted'
```

Comment:
I suggest consolidating the logic for the `.is_corrupted` extension (the pattern for the file path) into a single location. This would ensure that future developers only need to make changes in one place, reducing the risk of them modifying it in one location and forgetting to do so in another.

Reply:
Done (except tests)